src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,413 @@
+    end
+  end
+  
+  # Query and process remaining rows
+  table_result = DBInterface.execute(db, "SELECT * FROM \"$table_name\"")
+  row_count = 0
+  for row in Tables.rows(table_result)
+    row_count += 1
+    if row_count == 1
+      continue  # Skip first row since we already processed it
+    end
+    
+    for col_name in all_columns
+      col_symbol = Symbol(col_name)
+      if col_name == geom_column
+        # Parse geometry blob
+        blob = Tables.getcolumn(row, col_symbol)
+        if ismissing(blob) || isnothing(blob)
+          push!(column_data[col_symbol], blob)
+        else
+          push!(column_data[col_symbol], parse_wkb_geometry_from_blob(blob, crs_type))
+        end
+      else
+        # Store attribute data as-is
+        push!(column_data[col_symbol], Tables.getcolumn(row, col_symbol))
+      end
+    end
+  end
+  
+  # 4. Build final table
+  final_names = Symbol[]
+  final_columns = Any[]
+  
+  for col_name in all_columns
+    if col_name == geom_column
+      push!(final_names, :geometry)
+    else
+      push!(final_names, Symbol(col_name))
+    end
+    push!(final_columns, column_data[Symbol(col_name)])
+  end
+  
+  close(db)
+  
+  return (; zip(final_names, final_columns)...)
+end
+
+function gpkgmetadata(db::SQLite.DB, layer::Int)
+  # Get feature table names
+  contents_result = DBInterface.execute(db,
+    """
+    SELECT table_name, data_type
+    FROM gpkg_contents
+    WHERE data_type = 'features'
+    ORDER BY table_name
+    """)
+  
+  # Use Iterators.drop to get the requested layer
+  rows = Tables.rows(contents_result)
+  layer_iter = Iterators.drop(rows, layer - 1)
+  
+  if isempty(layer_iter)
+    error("Layer $layer not found in GeoPackage")
+  end
+  first_row = first(layer_iter)
+  
+  table_name = first_row.table_name
+  
+  # Get geometry column information
+  geom_result = DBInterface.execute(db,
+    """
+    SELECT column_name, srs_id
+    FROM gpkg_geometry_columns
+    WHERE table_name = ?
+    """, [table_name])
+  geom_info = first(Tables.rows(geom_result))
+  geom_column = geom_info.column_name
+  srs_id = geom_info.srs_id
+  
+  # Get CRS from SRID according to GeoPackage specification
+  # Handle special cases per GeoPackage spec
+  if srs_id == 0
+    crs_type = LatLon{WGS84Latest}
+  elseif srs_id == -1
+    # Undefined Cartesian coordinate reference system - determine dimensionality from geometry_columns
+    geom_cols_result = DBInterface.execute(db,
+      """
+      SELECT z
+      FROM gpkg_geometry_columns
+      WHERE table_name = ? AND column_name = ?
+      """, [table_name, geom_column])
+    
+    geom_state = iterate(Tables.rows(geom_cols_result))
+    if !isnothing(geom_state)
+      z_flag = geom_state[1].z
+      if z_flag == 1
+        crs_type = Cartesian{NoDatum,3}
+      else
+        crs_type = Cartesian{NoDatum,2}
+      end
+    else
+      crs_type = Cartesian{NoDatum,2}
+    end
```

Comment:
Why do we need to extract the dimensionality here? Can't we just return Cartesian{NoDatum} like it is done below?